src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -392,10 +392,12 @@ private void validatePartitionProperties(FailureCollector collector) {
     if (sourceTable == null) {
       return;
     }
-    TimePartitioning timePartitioning = ((StandardTableDefinition) (sourceTable).getDefinition())
-      .getTimePartitioning();
-    if (timePartitioning == null) {
-      return;
+    if (sourceTable.getDefinition() instanceof StandardTableDefinition) {
+      TimePartitioning timePartitioning = ((StandardTableDefinition) sourceTable.getDefinition())
+          .getTimePartitioning();
```

Comment:
indentation is off (should be 2 spaces). Please make sure your IDE has the right settings.